File: python_MVC/dao/daoImpl/PopImpl.py
# daoImpl/popImpl.py

from dao.Pop import Pop
from entity.Filed import Filed

class PopImpl(Pop):
    def pop(self, stack: Filed) -> int:
        if stack.getSp() > 0:
            value = stack.getStack()[stack.getSp()]
            stack.setSp(stack.getSp() - 1)
            return value
        else:
            print("スタックが空です")
            return -999  # エラー値を返す

File: java_MVC/dao/daoImpl/PopImpl.java
package dao.daoImpl;

import dao.Pop;
import entity.Filed;

public class PopImpl implements Pop {

    @Override
    public int pop(Filed stack) {
        if (stack.getSp() > 0) {
                        //Filedクラスの変数に
            int value = stack.getStack()[stack.getSp() - 1];
            stack.setSp(stack.getSp() - 1);
            return value;
        } else {
            System.out.println("スタックが空です");
            return -999; // エラー値を返す
        }
    }
}